include/networkit/centrality/GedWalk.hpp

```diff
@@ -0,0 +1,220 @@
+        walks w;
+    };
+
+    tlx::d_ary_addressable_int_heap<node, 2, CompareScore> scoreQ{CompareScore(gainScore)};
+    tlx::d_ary_addressable_int_heap<node, 2, CompareScore> boundQ{CompareScore(gainBound)};
+
+    void init();
+    double computeGamma();
+    void estimateGains();
+    void computeMarginalGain(node z);
+    void maximizeGain();
+    bool separateNodes();
+    double computeSigmaMax() const;
+    void fillPQs();
+    void updateAlphas();
+    EvaluationResult evaluateGraph();
+    EvaluationResult evaluateGroup();
+
+public:
+    /**
+     * Finds a group of `k` vertices with at least ((1 - 1/e) * opt - epsilon) GedWalk centrality
+     * score, where opt is the highest possible score. The algorithm is based on the paper "Group
+     * Centrality Maximization for Large-scale Graphs", Angriman et al., ALENEX20. It implements two
+     * independent greedy strategies (lazy and stochastic). Furthermore, it allows to compute the
+     * GedWalk score of a given set of nodes.
+     *
+     * @param G A (weakly) connected graph.
+     * @param k The desired group size.
+     * @param epsilon Precision of the algorithm.
+     * @param alpha Exponent to compute the GedWalk score.
+     * @param bs Bound strategy to compute the GedWalk bounds, default: BoundStrategy::geometric.
+     * @param gs Greedy strategy to be used (lazy or stochastic), default: GreedyStrategy::lazy.
+     * @param spectralDelta Delta to be used for the spectral bound.
+     *
+     * Note: if the graph is weighted, all weights should be in (0, 1].
+     */
+    GedWalk(const Graph &G, count k = 1, double initEpsilon = 0.1, double alpha = -1.,
+            BoundStrategy bs = BoundStrategy::geometric, GreedyStrategy gs = GreedyStrategy::lazy,
+            double spectralDelta = 0.5);
+
+    /**
+     * Run the algorithm.
+     */
+    void run() override;
+
+    /**
+     * Return the approximate GedWalk score of the computed group.
+     *
+     * @return The approximate score of the computed group.
+     */
+    double getApproximateScore() const {
+        assureFinished();
+        return groupScore;
+    }
+
+    /**
+     * Return the GedWalk score of the input group.
+     *
+     * @param first/last The range that contains the input group.
+     * @param scoreEpsilon The precision of the score to be computed.
+     */
+    template <class InputIt>
+    double scoreOfGroup(InputIt first, InputIt last, double scoreEpsilon = .1);
+
+    /**
+     * Return the computed group.
+     *
+     * @return The computed group.
+     */
+    std::vector<node> groupMaxGedWalk() const {
+        assureFinished();
+        return group;
+    }
+
+    const std::vector<double> &boundOfMarginalGains() const { return gainBound; }
+};
+
+template <class InputIt>
+double GedWalk::scoreOfGroup(InputIt first, InputIt last, double scoreEpsilon) {
+    if (boundStrategy == BoundStrategy::spectral) {
+        assert(alpha < 1 / static_cast<double>(sigmaMax));
```

Comment:
Only a very small remark: Depending on the value of `sigmaMax` and `degInMax`, the assertion can benefit (both in precision and clocktime) if compare `alpha * static_cast<double>(something)` to `1`.